### PR TITLE
Add `ignore_transfer_host` to ignore data channel host

### PR DESCRIPTION
**Pull request by Frankkkkk**
Hi,

Some servers (or setups) send misconfigured data channel hosts. We
add a way to bypass these hosts and thus use the original server IP.

Signed-off-by: Frank Villaro-Dixon <frank.villaro@infomaniak.com>

### CODE REVIEW

**Comment by pyama86** [comment]
Thanks for your contribute.
I understood your idea and I want to improve this PR.
so, you sometime wrote same code.

```go
if !isPublicIP(net.ParseIP(d.clientConn.remoteIP)) || d.config.IgnoreTransferHost {
```
I think you had better to make new function and add test if you have a time.

thanks.

**Comment by Frankkkkk** [comment]
Indeed, It would be better than a simple copy/paste 👍  I'll try to modify the PR tomorrow
Cheers

**Comment by Frankkkkk** [comment]
Hi @pyama86 , I had things mixed up yesterday. Indeed the "ignore" logic must only happen in the responses from the FTP server and not for the commands sent to it (i.e. only for passive and not active).

Here's the updated MR. Finally the change is way less copy/paste than before.

I've added 2 tests + fixed them because the struct check was not done before. I'll open another MR for the other ones.

PS: could you please not squash/rebase my merge request please ?

Thanks and cheers !

File: pftp/data_handler_test.go [hunk 1]
```diff
@@ -301,14 +301,44 @@ func Test_dataHandler_parsePASVresponse(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "passive_mode_parse_ok",
+			name: "passive_mode_parse_ok_public",
 			fields: fields{
-				line:   "227 Entering Passive Mode (10,19,10,10,100,10).\r\n",
+				line:   "227 Entering Passive Mode (20,30,40,50,100,10).\r\n",
```
Comment (pyama86) [inline pftp/data_handler_test.go:306]: Why did you need to change an existing test?

Comment (Frankkkkk) [reply pftp/data_handler_test.go:306]: Because the existing one didn't work: https://github.com/pyama86/pftp/blob/master/pftp/data_handler_test.go#L306 We sent `10,19,10,10` (< typo 19 instead of 10) and checked for `10.10.10.10`. Also as we ignore private IPs by default it was not used.

The logic of the existing test is now called `passive_mode_parse_ok_private`

Comment (pyama86) [reply pftp/data_handler_test.go:306]: I see.
To be honestry,I don't use this software myself for a long time.
However,my colleagues continue using and I shoud maintain this repository.
I habe added GitHub Actions to in this repository and I hope  keep quority this repository.
thanks for you contribute.

Comment (pyama86) [reply pftp/data_handler_test.go:306]: could you please pull origin master!

Comment (Frankkkkk) [reply pftp/data_handler_test.go:306]: done

**Comment by pyama86** [comment]
thanks!
